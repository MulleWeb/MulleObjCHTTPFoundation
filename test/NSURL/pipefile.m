//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//

#ifdef __MULLE_OBJC__
# import <MulleObjCInetFoundation/MulleObjCInetFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif

//#import "MulleStandaloneObjCFoundation.h"

char  *safe_string( char *s)
{
   return( s ? s : "*null*");
}


int main( int argc, const char * argv[])
{
   NSURL   *url;
   char    *a, *b;

   url = [NSURL URLWithString:@"/|.scion"];
   if( url)
   {
      a = safe_string( [[url host] UTF8String]);
      b = safe_string( [[url path] UTF8String]);
      printf( "host:%s path:%s\n", a, b);
   }
   else
      printf( "invalid #1\n");

   url = [NSURL URLWithString:@"|.scion"];
   if( url)
   {
      a = safe_string( [[url host] UTF8String]);
      b = safe_string( [[url path] UTF8String]);
      printf( "host:%s path:%s\n", a, b);
   }
   else
      printf( "invalid #2\n");

   return( 0);
}
