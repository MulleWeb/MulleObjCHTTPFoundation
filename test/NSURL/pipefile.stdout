invalid #1
invalid #2
